### PR TITLE
fix: check existing idp

### PR DESCRIPTION
**Pull request by fgerschwiler**

### CODE REVIEW

File: internal/iam/repository/eventsourcing/iam.go [hunk 1]
```diff
@@ -316,30 +317,48 @@ func checkExistingLoginPolicyIDPProviderValidation(idpConfigID string) func(...*
 			switch event.Type {
 			case model.IDPConfigAdded:
 				config := new(model.IDPConfig)
-				config.SetData(event)
+				err := config.SetData(event)
+				if err != nil {
+					logging.LogWithFields("ERROR-6Hs9d", "event", event).WithError(err).Error("could not set data")
```
Comment (adlerhurst) [inline internal/iam/repository/eventsourcing/iam.go:322]: logging the whole event is too much

File: internal/iam/repository/eventsourcing/iam.go [hunk 2]
```diff
@@ -316,30 +317,48 @@ func checkExistingLoginPolicyIDPProviderValidation(idpConfigID string) func(...*
...
+				}
 				idpConfigs = append(idpConfigs, config)
 			case model.IDPConfigRemoved:
 				config := new(model.IDPConfig)
-				config.SetData(event)
-				for i, p := range idpConfigs {
-					if p.IDPConfigID == config.IDPConfigID {
+				err := config.SetData(event)
+				if err != nil {
+					logging.LogWithFields("ERROR-fGj8s", "event", event).WithError(err).Error("could not set data")
```
Comment (adlerhurst) [inline internal/iam/repository/eventsourcing/iam.go:330]: logging the whole event is too much

File: internal/iam/repository/eventsourcing/iam.go [hunk 3]
```diff
@@ -316,30 +317,48 @@ func checkExistingLoginPolicyIDPProviderValidation(idpConfigID string) func(...*
...
 						idpConfigs = idpConfigs[:len(idpConfigs)-1]
+						break
 					}
 				}
 			case model.LoginPolicyIDPProviderAdded:
 				idp := new(model.IDPProvider)
-				idp.SetData(event)
+				err := idp.SetData(event)
+				if err != nil {
+					logging.LogWithFields("ERROR-6Hjs8", "event", event).WithError(err).Error("could not set data")
```
Comment (adlerhurst) [inline internal/iam/repository/eventsourcing/iam.go:345]: logging the whole event is too much

File: internal/iam/repository/eventsourcing/iam.go [hunk 4]
```diff
@@ -316,30 +317,48 @@ func checkExistingLoginPolicyIDPProviderValidation(idpConfigID string) func(...*
...
+				}
 				idps = append(idps, idp)
 			case model.LoginPolicyIDPProviderRemoved:
 				idp := new(model.IDPProvider)
-				idp.SetData(event)
-				for i, p := range idps {
-					if p.IDPConfigID == idp.IDPConfigID {
+				err := idp.SetData(event)
+				if err != nil {
+					logging.LogWithFields("ERROR-6fh8s", "event", event).WithError(err).Error("could not set data")
```
Comment (adlerhurst) [inline internal/iam/repository/eventsourcing/iam.go:353]: logging the whole event is too much

File: internal/org/repository/eventsourcing/login_policy.go [hunk 5]
```diff
@@ -127,25 +131,38 @@ func checkExistingLoginPolicyIDPProviderValidation(idpProvider *iam_es_model.IDP
...
+						break
 					}
 				}
 			case model.LoginPolicyIDPProviderAdded:
 				idp := new(iam_es_model.IDPProvider)
-				idp.SetData(event)
-			case model.LoginPolicyIDPProviderRemoved:
+				err := idp.SetData(event)
+				if err != nil {
+					logging.LogWithFields("ERROR-6Hjs8", "event", event).WithError(err).Error("could not set data")
```
Comment (adlerhurst) [inline internal/org/repository/eventsourcing/login_policy.go:150]: logging the whole event is too much

File: build/docker/Dockerfile [hunk 6]
```diff
@@ -1,14 +1,14 @@
 # This Stage prepares the user in the container and copies the files
```
Comment (adlerhurst) [inline build/docker/Dockerfile:1]: you really wanted to change this file?

**Review by adlerhurst — CHANGES_REQUESTED** [review]

File: internal/iam/repository/eventsourcing/iam.go (hunk 1 above)
Comment (fgerschwiler) [reply internal/iam/repository/eventsourcing/iam.go]: removed

File: internal/iam/repository/eventsourcing/iam.go (hunk 2 above)
Comment (fgerschwiler) [reply internal/iam/repository/eventsourcing/iam.go]: removed

File: internal/iam/repository/eventsourcing/iam.go (hunk 4 above)
Comment (fgerschwiler) [reply internal/iam/repository/eventsourcing/iam.go]: removed

File: internal/iam/repository/eventsourcing/iam.go (hunk 3 above)
Comment (fgerschwiler) [reply internal/iam/repository/eventsourcing/iam.go]: removed

File: internal/org/repository/eventsourcing/login_policy.go (hunk 5 above)
Comment (fgerschwiler) [reply internal/org/repository/eventsourcing/login_policy.go]: removed

File: build/docker/Dockerfile (hunk 6 above)
Comment (fforootd) [reply build/docker/Dockerfile:1]: yes, because otherwise it will not work due to #810

**Review by fforootd — APPROVED** [review]

**Review by adlerhurst — APPROVED** [review]

**Pull request merged**